TF-3649 turn no-non-zero-settimeout into no-settimeout

joecan opened the pull request:
##### *To:*
@jjinux @kevinzang @cansin @inancsevinc 

##### *What:*
Since we are moving to the async / await this rule will check for `setTimeout` usages and suggest `await delay` instead. A fix is not possible.

##### *JIRA:*
https://udemyjira.atlassian.net/browse/TF-3649

##### *What did you test:*
yarn test

##### *What dashboards will you be monitoring:*
N/A

joecan committed:
TF-3649 turn no-non-zero-settimeout into no-settimeout

jjinux commented on packages/eslint-plugin-udemy/rules/no-settimeout/README.md:
```diff
@@ -1,6 +1,6 @@
+# No setTimeout
 
-`settimeout/no-nonzero` rule checks for any uses of `window.setTimeout` with a non-zero delay value.
+`udemy/no-settimeout` rule checks for any uses of `window.setTimeout.
```
It seems like you're missing a backtick.

joecan replied:
🤦Fixed.

jjinux commented on packages/eslint-plugin-udemy/rules/no-settimeout/README.md:
````diff
@@ -9,6 +9,13 @@ This rule takes no arguments.
 The following patterns are considered warnings:
 
 ```js
+setTimeout();
````
I'm guessing you can probably delete all but 2 of these examples:

 * setTimeout(...);
 * window.setTimeout(...);

For the tests, you can get a little more specific, but you probably don't even need to given how the code works.

joecan replied:
okay, I actually had to alter my test to catch window.setTimeout()

jjinux commented on packages/eslint-plugin-udemy/rules/no-settimeout/index.js:
```diff
@@ -0,0 +1,18 @@
+                    if (node.expression.callee && node.expression.callee.name === 'setTimeout') {
+                        context.report({
+                            node,
+                            message: 'Found `setTimeout` used in a spec. Please use `await delay` instead.',
```
This says "spec", but there's nothing about the rule that is specific to specs, right?

jjinux replied:
Should there be parentheses after `delay()`? Otherwise, the code will actually work, but it'll be subtly broken.

joecan replied:
Okay, I fixed the wording of the error report.

jjinux reviewed (approved):
You might want to rebase your commit in order to change your commit message to follow the Conventional Commits standard per (https://github.com/udemy/js-tooling). This is a breaking change since you're removing a rule.

joecan committed:
Publish

 - babel-polyfill-udemy-website@8.0.1
 - babel-preset-udemy-website@9.0.1
 - eslint-config-udemy-basics@6.0.1
 - eslint-config-udemy-website@8.0.1
 - eslint-plugin-udemy@7.0.1
